app/models/driver.rb

```diff
@@ -0,0 +1,49 @@
+require 'money'
+class Driver < ApplicationRecord
+  has_many :trips
+
+  validates :name, presence: true
+  validates :vin, presence: true
+
+  attribute :available, default: true
+
+  def mark_available
+    if self.available
+      self.available = false
+    else
+      self.available = true
+    end
+    self.save
+  end
+#  drivers total earnings
+  #  driver gets 80% of the trip cost after a fee of $1.65 is subtracted
+  def total_revenue
+    return 0 if self.trips.length == 0
+
+    commission = 0.8
+    fee = 165
+    total = 0
+    self.trips.each do |trip|
+      if trip.cost > fee
+        total += (trip.cost - fee) * commission
+      else
+        total += (trip.cost * commission)
+      end
+    end
+    return Money.new(total.ceil, "USD").format
+  end
+
+  def average_rating
+    total_rating = 0.0
+    rated_trips = 0
+
+    self.trips.each do |trip|
+      if trip.rating != nil
+        total_rating += trip.rating
+        rated_trips += 1
+      end
+    end
```

Comment:
Awesome logic! I see what you're going for here with counting the rated_trips.

Your code inspires me to suggest another refactoring: What if you made an array of only the rated trips, by using [the filter method](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-filter), and then summed the ratings together? Then, you wouldn't need to check if `trip.rating` is `nil`, and you wouldn't have to count `rated_trips`.

Again, this is a totally optional refactoring, I'm just suggesting it to help you think of cleaner code in the future :)